In [2]:
import numpy as np
import pandas as pd
import requests

In [19]:
from urllib.request import urlopen
from bs4 import BeautifulSoup
import ssl
import csv

In [8]:
ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

In [84]:
# acquire data
source = requests.get('http://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M').text
soup = BeautifulSoup(source, 'lxml')

In [85]:
# view the table attribute and substract it
my_table = soup.find('table', {'class' : 'wikitable'})
# seperate rows through 'tr'
table_rows = my_table.find_all('tr')

In [86]:
# separate items through 'td'
data = []
for row in table_rows:
    data.append([t.text.strip() for t in row.find_all('td')])
df = pd.DataFrame(data, columns=['PostalCode', 'Borough', 'Neighbourhood'])
# filter out bad rows
df = df[~df['PostalCode'].isnull()]  

In [89]:
df['Neighbourhood'] = df['Neighbourhood'].str.replace('/', ',')

In [90]:
df = df.loc[df['Borough'] != 'Not assigned',]

In [92]:
df.head()

PostalCode           Borough                                 Neighbourhood
3        M3A        North York                                     Parkwoods
4        M4A        North York                              Victoria Village
5        M5A  Downtown Toronto                    Regent Park , Harbourfront
6        M6A        North York             Lawrence Manor , Lawrence Heights
7        M7A  Downtown Toronto  Queen's Park , Ontario Provincial Government

In [91]:
df.shape

(103, 3)

In [83]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 103 entries, 3 to 179
Data columns (total 3 columns):
PostalCode       103 non-null object
Borough          103 non-null object
Neighbourhood    103 non-null object
dtypes: object(3)
memory usage: 3.2+ KB


#### In total, there are 103 postal codes extracted from the table in wikipedia, that means there are 103 postal codes in Canada.